### DISCUSSION TITLE
I2S(RX)  to  Multioutput  (TX)-> A2DP, I2X , Serial  Simultaneously

### DISCUSSION THREAD

**Discussion by vednildip** [pschatzmann/arduino-audio-tools#1812] [Q&A] [answered]
Implementation of   I2S(RX) sampling@( 44100, 16 bit , 2)  to  Multioutput  (TX)-> A2DP, I2X , Serial  Simultaneously    but   the  output  is glitches , dropping  and  noisy ; Where  as  calling  individual  output  working  very  smoothly in  each  case .

**Comment by pschatzmann** [comment, marked as answer]
Multioutput is obviously limiting the output to the slowest output component: for Serial not holding up the output you will need to send it with at least 44100 * (16 + 1) * 2 = 1499400 Baud (if you did not change the default config).

The I2S input and output is also not optimal together with A2DP since since it is of advantage to use a source and destination that can process the data faster then the related input/output rate (e.g. files).

**Comment by vednildip** [comment]
Thanks , even  I  Omit  Serial  then   how I2S and A2DP are handled to ensure the system can process them without glitches as Partially  understood  " source and destination concept"  but  seems beyond  my  C++expertise  about  code  implementation related input/output rate (e.g. files).
 Any  way  thanks I  will  ignore  serial .
